test/model.spec.js

```diff
@@ -600,5 +601,119 @@ describe('unit tests for model Objects', () => {
       const ilmd = new Ilmd(exampleIlmd);
       expect(ilmd.toObject()).to.deep.equal(exampleIlmd);
     });
+
+    describe('Entity.js', () => {
+      describe('generateSetterFunction', () => {
+        it('should throw an error if there is no expected type', async () => {
+          const field = 'ilmd';
+          const param = new Ilmd();
+          const o = new ObjectEvent();
+          assert.throws(() => o.generateSetterFunction(field, param, []));
```

Comment:
For visibility, you don't need to declare new variables to set things like this.
```
assert.throws(() => o.generateSetterFunction('ilmd', new Ilmd(), []));
```